feat(rust): show interactive menu to select one or more TCP inlent when user dosen't specifies

0x61nas opened the pull request:
Make ockam tcp-inlet delete (no args) interactive by asking the user to choose from a list of `tcp-inlet` aliases to delete using tuify

 After this PR gets merged and if no one takes (#6465, #6464, #6463 and #6462) I'll apply the same changes in one or multiple othr PRs to close them

 - Closes: build-trust/ockam/issues/6466

<!-- Thank you for sending a pull request :heart: -->

## Current behavior
Shows an error message when the user didn't specify an alias

<!-- Please describe the current behavior of the code before the changes in this pull request are applied. -->

## Proposed changes
- Make the `alias` argument optional
- Check if the user specifies an alias or not
- if so, then do the old behavior (check, confirm, delete)
- if not, and the terminal is interactive, and the user didn't pass any of `--no-output` or `--quiet` flags
  - then collect all exists TCP inlets aliases in the (selected or default) node
  - Shows them in a multi selection TUI menu
  - confirm and delete 

<!-- Please describe the changes proposed in this pull request. -->
<!-- If this pull request resolves an already recorded bug or a feature request, please add a link to that issue. -->

## Checks

<!--
To help us review and merge this pull request quickly, please confirm the following
by replacing the [ ] in front of each bullet point below with [x]
-->

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] There are no Merge commits in this Pull Request. Ockam repo maintains a linear commit history. We merge Pull Requests by rebasing them onto the develop branch. Rebasing to the latest develop branch and force pushing to your Pull Request branch is okay.
- [x] I have read and accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have read and accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/.github/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam/blob/develop/.github/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam](https://github.com/build-trust/ockam) repository. done via: build-trust/ockam-contributors/pull/266

<!-- We're looking forward to merging your contribution!! -->

adrianbenavides commented:
Hi @0x61nas , thanks for submitting this PR! We recently merged the first PR related to making some commands interactive at https://github.com/build-trust/ockam/pull/6480

That PR can serve as a guide on how to use some common functionalities that are now available as part of the `Terminal` struct. Make sure to rebase latest develop. And let me know if you have any questions!

0x61nas commented:
@adrianbenavides Hi, is there any plan to review this anytime soon(in this October)?. Hope you have a nice day.

adrianbenavides reviewed (commented):
It's looking great @0x61nas ! We have two similar PR's related to other `show` commands where we are using a different structure to handle the multiple use cases. My only request for now would be to adapt your PR to that structure (check https://github.com/build-trust/ockam/pull/6547 and https://github.com/build-trust/ockam/pull/6665). Let me know if you have any questions!

mrinalwadhwa commented:
@0x61nas while we still have to review this change. I'm adding the `HACKTOBERFEST-ACCEPTED` label so the contribution is counted by hacktoberfest bots before the end of October. Thank you for taking time to contribute 🎉 

Please not @adrianbenavides's feedback above.

0x61nas commented:
> It's looking great @0x61nas ! We have two similar PR's related to other `show` commands where we are using a different structure to handle the multiple use cases. My only request for now would be to adapt your PR to that structure (check #6547 and #6665). Let me know if you have any questions!

fine, i'll follow it

0x61nas commented on implementations/rust/ockam/ockam_command/src/tcp/util.rs:
```diff
@@ -8,3 +20,27 @@ pub fn alias_parser(arg: &str) -> Result<String> {
     }
 }
+
+pub async fn fetch_list<T: TerminalWriter, L: for<'b> minicbor::Decode<'b, ()>>(
```
I meant with this to use it in other commands to fetch the needed recourses list instead of duplicated code.
I wanna use it in the `inlet list` command and other list and delete commands. Do you think that there's a better way?

adrianbenavides replied:
You have an example here on how this was solved for the `node` commands: https://github.com/build-trust/ockam/pull/6665

0x61nas committed:
feat(rust): show interactive menu to select one or more TCP inlent when user dosen't specifies

  Make ockam tcp-inlet delete (no args) interactive by asking the user to choose from a list of `tcp-inlet` aliases to delete using tuify

 - Closes: build-trust/issues/6466

adrianbenavides commented:
Hi @0x61nas , your PR will be merged through https://github.com/build-trust/ockam/pull/6886 Thanks again for your help!
